carbon-design-system/carbon-addons-iot-react
# feat(sidepanel): extend component features

Closes #3497

**Summary**

Address gaps in SidePanel which were identified while incorporating SidePanel in Graphite's SlidingDrawer(default theme)

**Change List (commits, features, bugs, etc)**

- Style updates done to always dock the footer buttons at bottom of panel with `2px` spacing between the buttons.
- Added `isBusy` prop of type boolean. When true will disallow user to close the side panel until the ongoing operation is completed. New knob `isBusy` added to storybook to demonstrate the same.
- Added `disabled` prop to SidePanel's footer secondary button, `loading` prop to footer primary button and `disabled` prop to close togglebtn. When `busy` is set to true will make close(x) button and footer secondary button disabled along with loading state for footer primary button.
- Exposed footer primary button's disabled prop - `isPrimaryButtonDisabled`. To support cases where primary button to be disabled until all required fields are populated with a value.

**Acceptance Test (how to verify the PR)**

- Go to /?path=/story/1-watson-iot-sidepanel--default and verify the style changes.
- Verify the `isBusy` prop by toggling `isBusy` knob.
- Verify `isPrimaryButtonDisabled` prop using the knob - `Footer primary button disabled`
- Go to /?path=/story/1-watson-iot-sidepanel--in-page-example and verify the style changes. Verify the panel buttons using `isBusy` and `Footer primary button disabled` knobs.

**Regression Test (how to make sure this PR doesn't break old functionality)**

- Go to /?path=/story/1-watson-iot-sidepanel--default and verify all functionalities should work as before.
- Go to /?path=/story/1-watson-iot-sidepanel--in-page-example and verify all functionalities should work as before.



**Things to look for during review**
